device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -763,9 +763,9 @@ private void commonConstructorVerification(IotHubConnectionString connectionStri
         }
 
         String gatewayHostName = connectionString.getGatewayHostName();
-        if (gatewayHostName != null && !gatewayHostName.isEmpty() && !(protocol == MQTT || protocol == MQTT_WS))
+        if (gatewayHostName != null && !gatewayHostName.isEmpty() && !(protocol == MQTT || protocol == MQTT_WS || protocol == AMQPS || protocol == AMQPS_WS))
```

Comment:
nit: only not support on HTTP.  Should we check for HTTP instead?

Reply:
sure

Reply:
missed?


---
In reply to: [241218891](https://github.com/Azure/azure-iot-sdk-java/pull/398#discussion_r241218891) [](ancestors = 241218891)